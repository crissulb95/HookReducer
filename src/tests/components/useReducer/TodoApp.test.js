const { act } = require("@testing-library/react");
const { shallow, mount } = require("enzyme");
const { TodoAdd } = require("../../../components/useReducer/TodoAdd");
const { TodoApp } = require("../../../components/useReducer/TodoApp");
const { demos } = require("../../fixtures/demoTodos");

describe('Pruebas en TodoApp', () => {
    
    const wrapper = shallow( <TodoApp />);

    Storage.prototype.setItem = jest.fn(() => {});

    test('Debe mostrarse bien ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe agregar un Todo', () => {

        const wrapper = mount(<TodoApp />);

        act( () => {
            wrapper.find('TodoAdd').prop('handleAddTodo')(demos[0])
            wrapper.find('TodoAdd').prop('handleAddTodo')(demos[1])
        } );

        expect(wrapper.find('h1').text().trim()).toBe('To Do\'s App (2) ');
        expect(localStorage.setItem).toHaveBeenCalledTimes(2)
        //Parece ser que esta versi√≥n de enzyme no funciona con mi react 17, por lo que hay que 
        //esperar a que salga un nuevo adaptador oficial para que logre funcionar la prueba
        //que da el error en la funcion mount como TypeError: Cannot read property 'child' of undefined

    });
    

})
