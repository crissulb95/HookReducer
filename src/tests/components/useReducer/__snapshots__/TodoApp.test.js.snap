// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en TodoApp Debe mostrarse bien  1`] = `
<div>
  <h1>
    To Do's App (
    0
    ) 
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-7"
    >
      <TodoList
        handleDelete={[Function]}
        handleTachado={[Function]}
        todos={Array []}
      />
    </div>
    <div
      className="col-5"
    >
      <TodoAdd
        handleAddTodo={[Function]}
      />
    </div>
  </div>
</div>
`;
